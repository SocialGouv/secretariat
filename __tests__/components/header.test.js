import { useSession } from "next-auth/react"
import { render } from "@testing-library/react"

import Header from "../../src/components/header"

// https://github.com/nextauthjs/next-auth/issues/775
jest.mock("next-auth/react", () => ({
  useSession: jest.fn(),
}))

Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
})

it("renders header", () => {
  useSession.mockReturnValueOnce([false, false])
  const { container } = render(<Header />)
  expect(container).toMatchSnapshot()
})
