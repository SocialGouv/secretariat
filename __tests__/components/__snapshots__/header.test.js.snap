// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders header 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="container"
    >
      <a
        href="/"
      >
        <div
          class="logo"
        >
          République
          <br />
          Française
        </div>
        <div
          class="title"
        >
          <h1>
            Secretariat
          </h1>
          <p>
            Gestion des comptes de services @SocialGouv
          </p>
        </div>
      </a>
      <nav />
    </div>
  </header>
</div>
`;
